compiler/z/codegen/OMRLinkage.cpp

```diff
@@ -1980,16 +1990,20 @@ OMR::Z::Linkage::buildArgs(TR::Node * callNode, TR::RegisterDependencyConditions
             //add extra return register dependency
             killMask = self()->addFECustomizedReturnRegDependency(killMask, self(), resType, dependencies);
             break;
-         case TR::VectorInt8:
-         case TR::VectorInt16:
-         case TR::VectorInt32:
-         case TR::VectorInt64:
-         case TR::VectorDouble:
-            resultReg = self()->cg()->allocateRegister(TR_VRF);
-            self()->cg()->setRealRegisterAssociation(resultReg, self()->getVectorReturnRegister());
-            dependencies->addPostCondition(resultReg, self()->getVectorReturnRegister(),DefinesDependentRegister);
-            killMask &= (~(0x1L << REGINDEX(self()->getVectorReturnRegister())));
-            break;
+         default:
+            if (resDataType.isVector())
+               {
+               TR::DataType elementType = resDataType.getVectorElementType();
+               if (elementType == TR::Int8 || elementType == TR::Int16 ||
+                   elementType == TR::Int32 || elementType == TR::Int64 || elementType == TR::Double)
```

Comment:
I am trying to understand why we have omitted FloatVector originally from having in arguments (Probably as short and extended format support for Vector instructions were added later in to the architecture). 
I am adding a `TODO` for myself here to double check and will confirm if that is the case, we might be able to remove check for element type here.

Reply:
Thanks! In some case above I already treat VectorFloat as all other vector types. Please let me know if should revert that.

Reply:
@r30shah just a reminder to let me know  how to treat VectorFloat in Z linkages.

Reply:
Please NVM. I will preserve current behavior.

Reply:
Will preserve current behavior.